spring-xd-related-work.tex

```diff
@@ -26,10 +26,27 @@ \subsection{Spark}
 supports integration with Spark applications such as Spark Streaming, MLLib, and
 SparkSQL. Users familiar with Spark may implement the computation logic using
 Spark APIs in Java or Scala, and leave the orchestration to Spring XD.
-The Spring XD Spark module acts as the driver while the computation defined
-by the streaming application is executed in the Spark cluster. Driver failure
-is automatically handled by Spring XD -- the admin server will re-deploy a
-module from a failed container to another eligible container.
+
+\subsubsection{Integration with any Spark application}
+ A Spark Application can be deployed and launched from Spring XD as
+a \emph{batch} job. A Spring XD batch job submits the Spark application into
+Spark cluster manager using org.apache.spark.deploy.SparkSubmit. The main advantage
+of this approach is that one can launch a Spark application with specific criteria
+via Spring XD stream. For instance, a real time scoring algorithm through Spark MLlib
+job can be triggered based on the criteria from a regular \emph{XD stream} events.
```

Comment:
One  advantage of this approach is that one can launch a Spark application with specific criteria
via Spring XD stream. For instance, a real time scoring algorithm through Spark MLlib
job can be triggered based on the criteria from a regular \emph{XD stream} events.  Another advantage is after the job is complete a notification can be sent to a stream for processing.